### PR TITLE
Feature/header component

### LINKED ISSUE #3: Header Component
We want a header component for guests and users to navigate our application efficiently. 

For now it can look something like 
image

with Join and Login as the only options and a typography of `Talent` on the left side.

See: [Mui App Bar](https://material-ui.com/components/app-bar/)

furthermore, we want header to be able to toggle between light and dark theme, see [Switch](https://material-ui.com/components/switches/#switch), it just needs to fire this function [here](https://github.com/H-Richard/talent-frontend/blob/master/src/App.tsx#L14-L16)

Acceptance Tests:
- [x] Header is it's own component
- [x] Theme colors can be toggled

### PR DESCRIPTION
**Pull request by ivinable2000**
Added Header Component, with Typography of Talent on left side and two buttons - Login and Join that link to the /login and /signup page respectively.
Also added Dark Mode Switch to switch theme to Dark Mode

Closes #3 
Screen Shot 2020-07-23 at 11 26 08 PM (2)
Screen Shot 2020-07-23 at 11 26 23 PM 2

### CODE REVIEW

File: src/components/Header.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+import AppBar from '@material-ui/core/AppBar';
+import Toolbar from '@material-ui/core/Toolbar';
+import Typography from '@material-ui/core/Typography';
+import Button from '@material-ui/core/Button';
+import Switch from '@material-ui/core/Switch';
+import Grid from '@material-ui/core/Grid';
+
+interface ToggleMode {
+  ClickHandler: (event: React.ChangeEvent<HTMLInputElement>) => void
+}
```
Comment (H-Richard) [inline src/components/Header.tsx:14]: The main interface should just be named `Props` and in react, 99% of the time prop names are camel case, also `toggleTheme` is a much better name here

```suggestion
interface Props {
  toggleTheme: (event: React.ChangeEvent<HTMLInputElement>) => void
}
```

File: src/components/Header.tsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    boxShadow: 'none',
+  },
+  track: {
+    border: `1px solid ${theme.palette.grey[400]}`,
+    borderRadius: 26 / 2,
+    opacity: 1,
+    backgroundColor: theme.palette.common.white,
+  },
+  checked: {},
+}))(Switch);
```
Comment (H-Richard) [inline src/components/Header.tsx:47]: do we need this much styling just for the switch?

File: src/App.tsx [hunk 3]
```diff
@@ -37,6 +38,7 @@ function App() {
         <Route path="/login" exact render={() => (isLoggedIn() ? <Redirect to="/home" /> : <Login />)} />
       </Switch>
       <Modals />
+      <Header ClickHandler={toggleMode} />
```
Comment (H-Richard) [inline src/App.tsx:41]: We can put this Above the `<Switch></Switch>`

File: src/components/Header.tsx [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+  root: {
+    flexGrow: 1,
+  },
+  title: {
+    flexGrow: 1,
+    color: theme.palette.common.white,
+  },
+}));
+
+const Header = (Props: ToggleMode) => {
```
Comment (H-Richard) [inline src/components/Header.tsx:59]: One last thing, to comply with ts, we can give our functional components this type:
```suggestion
const Header: React.FC<T> = (Props: ToggleMode) => {
```

Where T is our Props interface

File: src/components/Header.tsx (hunk 2 above)
Comment (ivinable2000) [reply src/components/Header.tsx]: Its a custom switch but I can use a default one instead

Comment (H-Richard) [reply src/components/Header.tsx]: any chance we can make it look like the one [here](https://redux.js.org/)

Comment (ivinable2000) [reply src/components/Header.tsx]: Done

File: src/components/Header/index.tsx [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+    flexGrow: 1,
+    color: theme.palette.common.white,
+  },
+}));
+
+const Header: React.FC<Props> = ({ toggleTheme }: Props) => {
+  const classes = useStyles();
+  return (
+    <div>
+      <AppBar position="fixed">
```
Comment (H-Richard) [inline src/components/Header/index.tsx:75]: Let's give it `color="default"` so it stops using primary colors

File: src/components/Header/index.tsx [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+  },
+  thumb: {
+    width: 30,
+    height: 30,
+    backgroundColor: theme.palette.common.white,
+    border: 'solid',
+    borderColor: theme.palette.grey[800],
+  },
+  track: {
+    backgroundSize: '30px 28px',
```
Comment (H-Richard) [inline src/components/Header/index.tsx:33]: can we also make the track color a slightly darker one like the one in redux.js.org, also perhaps we can make the emoji not fill fit edge to edge, we can do something like `backgroundPosition: 'center', backgroundRepeat: 'no-repeat'` probably

File: src/components/Header/index.tsx [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+        backgroundColor: 'rgba(24,90,257,0.08)',
+      },
+    },
+    '& $thumb': {
+      backgroundColor: theme.palette.common.white,
+      border: 'solid',
+      borderColor: theme.palette.grey[800],
+    },
+    '& + $track': {
+      backgroundSize: '34px 28px',
```
Comment (H-Richard) [inline src/components/Header/index.tsx:55]: Same thing here, just not edge to edge would be good 👍

**Pull request merged**